// Code generated by smithy-swift-codegen. DO NOT EDIT!

import ClientRuntime

/// IAM Identity Center (successor to Single Sign-On) helps you securely create, or connect, your workforce identities and manage their access centrally across Amazon Web Services accounts and applications. IAM Identity Center is the recommended approach for workforce authentication and authorization in Amazon Web Services, for organizations of any size and type. IAM Identity Center uses the sso and identitystore API namespaces. This reference guide provides information on single sign-on operations which could be used for access management of Amazon Web Services accounts. For information about IAM Identity Center features, see the [IAM Identity Center User Guide](https://docs.aws.amazon.com/singlesignon/latest/userguide/what-is.html). Many operations in the IAM Identity Center APIs rely on identifiers for users and groups, known as principals. For more information about how to work with principals and principal IDs in IAM Identity Center, see the [Identity Store API Reference](https://docs.aws.amazon.com/singlesignon/latest/IdentityStoreAPIReference/welcome.html). Amazon Web Services provides SDKs that consist of libraries and sample code for various programming languages and platforms (Java, Ruby, .Net, iOS, Android, and more). The SDKs provide a convenient way to create programmatic access to IAM Identity Center and other Amazon Web Services services. For more information about the Amazon Web Services SDKs, including how to download and install them, see [Tools for Amazon Web Services](http://aws.amazon.com/tools/).
public protocol SSOAdminClientProtocol {
    /// Attaches the specified customer managed policy to the specified [PermissionSet].
    ///
    /// - Parameter AttachCustomerManagedPolicyReferenceToPermissionSetInput : [no documentation found]
    ///
    /// - Returns: `AttachCustomerManagedPolicyReferenceToPermissionSetOutputResponse` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : You do not have sufficient access to perform this action.
    /// - `ConflictException` : Occurs when a conflict with a previous successful write is detected. This generally occurs when the previous write did not have time to propagate to the host serving the current request. A retry (with appropriate backoff logic) is the recommended response to this exception.
    /// - `InternalServerException` : The request processing has failed because of an unknown error, exception, or failure with an internal server.
    /// - `ResourceNotFoundException` : Indicates that a requested resource is not found.
    /// - `ServiceQuotaExceededException` : Indicates that the principal has crossed the permitted number of resources that can be created.
    /// - `ThrottlingException` : Indicates that the principal has crossed the throttling limits of the API operations.
    /// - `ValidationException` : The request failed because it contains a syntax error.
    func attachCustomerManagedPolicyReferenceToPermissionSet(input: AttachCustomerManagedPolicyReferenceToPermissionSetInput) async throws -> AttachCustomerManagedPolicyReferenceToPermissionSetOutputResponse
    /// Attaches an Amazon Web Services managed policy ARN to a permission set. If the permission set is already referenced by one or more account assignments, you will need to call [ProvisionPermissionSet] after this operation. Calling ProvisionPermissionSet applies the corresponding IAM policy updates to all assigned accounts.
    ///
    /// - Parameter AttachManagedPolicyToPermissionSetInput : [no documentation found]
    ///
    /// - Returns: `AttachManagedPolicyToPermissionSetOutputResponse` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : You do not have sufficient access to perform this action.
    /// - `ConflictException` : Occurs when a conflict with a previous successful write is detected. This generally occurs when the previous write did not have time to propagate to the host serving the current request. A retry (with appropriate backoff logic) is the recommended response to this exception.
    /// - `InternalServerException` : The request processing has failed because of an unknown error, exception, or failure with an internal server.
    /// - `ResourceNotFoundException` : Indicates that a requested resource is not found.
    /// - `ServiceQuotaExceededException` : Indicates that the principal has crossed the permitted number of resources that can be created.
    /// - `ThrottlingException` : Indicates that the principal has crossed the throttling limits of the API operations.
    /// - `ValidationException` : The request failed because it contains a syntax error.
    func attachManagedPolicyToPermissionSet(input: AttachManagedPolicyToPermissionSetInput) async throws -> AttachManagedPolicyToPermissionSetOutputResponse
    /// Assigns access to a principal for a specified Amazon Web Services account using a specified permission set. The term principal here refers to a user or group that is defined in IAM Identity Center. As part of a successful CreateAccountAssignment call, the specified permission set will automatically be provisioned to the account in the form of an IAM policy. That policy is attached to the IAM role created in IAM Identity Center. If the permission set is subsequently updated, the corresponding IAM policies attached to roles in your accounts will not be updated automatically. In this case, you must call [ProvisionPermissionSet] to make these updates. After a successful response, call DescribeAccountAssignmentCreationStatus to describe the status of an assignment creation request.
    ///
    /// - Parameter CreateAccountAssignmentInput : [no documentation found]
    ///
    /// - Returns: `CreateAccountAssignmentOutputResponse` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : You do not have sufficient access to perform this action.
    /// - `ConflictException` : Occurs when a conflict with a previous successful write is detected. This generally occurs when the previous write did not have time to propagate to the host serving the current request. A retry (with appropriate backoff logic) is the recommended response to this exception.
    /// - `InternalServerException` : The request processing has failed because of an unknown error, exception, or failure with an internal server.
    /// - `ResourceNotFoundException` : Indicates that a requested resource is not found.
    /// - `ServiceQuotaExceededException` : Indicates that the principal has crossed the permitted number of resources that can be created.
    /// - `ThrottlingException` : Indicates that the principal has crossed the throttling limits of the API operations.
    /// - `ValidationException` : The request failed because it contains a syntax error.
    func createAccountAssignment(input: CreateAccountAssignmentInput) async throws -> CreateAccountAssignmentOutputResponse
    /// Enables the attributes-based access control (ABAC) feature for the specified IAM Identity Center instance. You can also specify new attributes to add to your ABAC configuration during the enabling process. For more information about ABAC, see [Attribute-Based Access Control] in the IAM Identity Center User Guide. After a successful response, call DescribeInstanceAccessControlAttributeConfiguration to validate that InstanceAccessControlAttributeConfiguration was created.
    ///
    /// - Parameter CreateInstanceAccessControlAttributeConfigurationInput : [no documentation found]
    ///
    /// - Returns: `CreateInstanceAccessControlAttributeConfigurationOutputResponse` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : You do not have sufficient access to perform this action.
    /// - `ConflictException` : Occurs when a conflict with a previous successful write is detected. This generally occurs when the previous write did not have time to propagate to the host serving the current request. A retry (with appropriate backoff logic) is the recommended response to this exception.
    /// - `InternalServerException` : The request processing has failed because of an unknown error, exception, or failure with an internal server.
    /// - `ResourceNotFoundException` : Indicates that a requested resource is not found.
    /// - `ThrottlingException` : Indicates that the principal has crossed the throttling limits of the API operations.
    /// - `ValidationException` : The request failed because it contains a syntax error.
    func createInstanceAccessControlAttributeConfiguration(input: CreateInstanceAccessControlAttributeConfigurationInput) async throws -> CreateInstanceAccessControlAttributeConfigurationOutputResponse
    /// Creates a permission set within a specified IAM Identity Center instance. To grant users and groups access to Amazon Web Services account resources, use [CreateAccountAssignment].
    ///
    /// - Parameter CreatePermissionSetInput : [no documentation found]
    ///
    /// - Returns: `CreatePermissionSetOutputResponse` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : You do not have sufficient access to perform this action.
    /// - `ConflictException` : Occurs when a conflict with a previous successful write is detected. This generally occurs when the previous write did not have time to propagate to the host serving the current request. A retry (with appropriate backoff logic) is the recommended response to this exception.
    /// - `InternalServerException` : The request processing has failed because of an unknown error, exception, or failure with an internal server.
    /// - `ResourceNotFoundException` : Indicates that a requested resource is not found.
    /// - `ServiceQuotaExceededException` : Indicates that the principal has crossed the permitted number of resources that can be created.
    /// - `ThrottlingException` : Indicates that the principal has crossed the throttling limits of the API operations.
    /// - `ValidationException` : The request failed because it contains a syntax error.
    func createPermissionSet(input: CreatePermissionSetInput) async throws -> CreatePermissionSetOutputResponse
    /// Deletes a principal's access from a specified Amazon Web Services account using a specified permission set. After a successful response, call DescribeAccountAssignmentDeletionStatus to describe the status of an assignment deletion request.
    ///
    /// - Parameter DeleteAccountAssignmentInput : [no documentation found]
    ///
    /// - Returns: `DeleteAccountAssignmentOutputResponse` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : You do not have sufficient access to perform this action.
    /// - `ConflictException` : Occurs when a conflict with a previous successful write is detected. This generally occurs when the previous write did not have time to propagate to the host serving the current request. A retry (with appropriate backoff logic) is the recommended response to this exception.
    /// - `InternalServerException` : The request processing has failed because of an unknown error, exception, or failure with an internal server.
    /// - `ResourceNotFoundException` : Indicates that a requested resource is not found.
    /// - `ThrottlingException` : Indicates that the principal has crossed the throttling limits of the API operations.
    /// - `ValidationException` : The request failed because it contains a syntax error.
    func deleteAccountAssignment(input: DeleteAccountAssignmentInput) async throws -> DeleteAccountAssignmentOutputResponse
    /// Deletes the inline policy from a specified permission set.
    ///
    /// - Parameter DeleteInlinePolicyFromPermissionSetInput : [no documentation found]
    ///
    /// - Returns: `DeleteInlinePolicyFromPermissionSetOutputResponse` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : You do not have sufficient access to perform this action.
    /// - `ConflictException` : Occurs when a conflict with a previous successful write is detected. This generally occurs when the previous write did not have time to propagate to the host serving the current request. A retry (with appropriate backoff logic) is the recommended response to this exception.
    /// - `InternalServerException` : The request processing has failed because of an unknown error, exception, or failure with an internal server.
    /// - `ResourceNotFoundException` : Indicates that a requested resource is not found.
    /// - `ThrottlingException` : Indicates that the principal has crossed the throttling limits of the API operations.
    /// - `ValidationException` : The request failed because it contains a syntax error.
    func deleteInlinePolicyFromPermissionSet(input: DeleteInlinePolicyFromPermissionSetInput) async throws -> DeleteInlinePolicyFromPermissionSetOutputResponse
    /// Disables the attributes-based access control (ABAC) feature for the specified IAM Identity Center instance and deletes all of the attribute mappings that have been configured. Once deleted, any attributes that are received from an identity source and any custom attributes you have previously configured will not be passed. For more information about ABAC, see [Attribute-Based Access Control] in the IAM Identity Center User Guide.
    ///
    /// - Parameter DeleteInstanceAccessControlAttributeConfigurationInput : [no documentation found]
    ///
    /// - Returns: `DeleteInstanceAccessControlAttributeConfigurationOutputResponse` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : You do not have sufficient access to perform this action.
    /// - `ConflictException` : Occurs when a conflict with a previous successful write is detected. This generally occurs when the previous write did not have time to propagate to the host serving the current request. A retry (with appropriate backoff logic) is the recommended response to this exception.
    /// - `InternalServerException` : The request processing has failed because of an unknown error, exception, or failure with an internal server.
    /// - `ResourceNotFoundException` : Indicates that a requested resource is not found.
    /// - `ThrottlingException` : Indicates that the principal has crossed the throttling limits of the API operations.
    /// - `ValidationException` : The request failed because it contains a syntax error.
    func deleteInstanceAccessControlAttributeConfiguration(input: DeleteInstanceAccessControlAttributeConfigurationInput) async throws -> DeleteInstanceAccessControlAttributeConfigurationOutputResponse
    /// Deletes the permissions boundary from a specified [PermissionSet].
    ///
    /// - Parameter DeletePermissionsBoundaryFromPermissionSetInput : [no documentation found]
    ///
    /// - Returns: `DeletePermissionsBoundaryFromPermissionSetOutputResponse` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : You do not have sufficient access to perform this action.
    /// - `InternalServerException` : The request processing has failed because of an unknown error, exception, or failure with an internal server.
    /// - `ResourceNotFoundException` : Indicates that a requested resource is not found.
    /// - `ThrottlingException` : Indicates that the principal has crossed the throttling limits of the API operations.
    /// - `ValidationException` : The request failed because it contains a syntax error.
    func deletePermissionsBoundaryFromPermissionSet(input: DeletePermissionsBoundaryFromPermissionSetInput) async throws -> DeletePermissionsBoundaryFromPermissionSetOutputResponse
    /// Deletes the specified permission set.
    ///
    /// - Parameter DeletePermissionSetInput : [no documentation found]
    ///
    /// - Returns: `DeletePermissionSetOutputResponse` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : You do not have sufficient access to perform this action.
    /// - `ConflictException` : Occurs when a conflict with a previous successful write is detected. This generally occurs when the previous write did not have time to propagate to the host serving the current request. A retry (with appropriate backoff logic) is the recommended response to this exception.
    /// - `InternalServerException` : The request processing has failed because of an unknown error, exception, or failure with an internal server.
    /// - `ResourceNotFoundException` : Indicates that a requested resource is not found.
    /// - `ThrottlingException` : Indicates that the principal has crossed the throttling limits of the API operations.
    /// - `ValidationException` : The request failed because it contains a syntax error.
    func deletePermissionSet(input: DeletePermissionSetInput) async throws -> DeletePermissionSetOutputResponse
    /// Describes the status of the assignment creation request.
    ///
    /// - Parameter DescribeAccountAssignmentCreationStatusInput : [no documentation found]
    ///
    /// - Returns: `DescribeAccountAssignmentCreationStatusOutputResponse` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : You do not have sufficient access to perform this action.
    /// - `InternalServerException` : The request processing has failed because of an unknown error, exception, or failure with an internal server.
    /// - `ResourceNotFoundException` : Indicates that a requested resource is not found.
    /// - `ThrottlingException` : Indicates that the principal has crossed the throttling limits of the API operations.
    /// - `ValidationException` : The request failed because it contains a syntax error.
    func describeAccountAssignmentCreationStatus(input: DescribeAccountAssignmentCreationStatusInput) async throws -> DescribeAccountAssignmentCreationStatusOutputResponse
    /// Describes the status of the assignment deletion request.
    ///
    /// - Parameter DescribeAccountAssignmentDeletionStatusInput : [no documentation found]
    ///
    /// - Returns: `DescribeAccountAssignmentDeletionStatusOutputResponse` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : You do not have sufficient access to perform this action.
    /// - `InternalServerException` : The request processing has failed because of an unknown error, exception, or failure with an internal server.
    /// - `ResourceNotFoundException` : Indicates that a requested resource is not found.
    /// - `ThrottlingException` : Indicates that the principal has crossed the throttling limits of the API operations.
    /// - `ValidationException` : The request failed because it contains a syntax error.
    func describeAccountAssignmentDeletionStatus(input: DescribeAccountAssignmentDeletionStatusInput) async throws -> DescribeAccountAssignmentDeletionStatusOutputResponse
    /// Returns the list of IAM Identity Center identity store attributes that have been configured to work with attributes-based access control (ABAC) for the specified IAM Identity Center instance. This will not return attributes configured and sent by an external identity provider. For more information about ABAC, see [Attribute-Based Access Control] in the IAM Identity Center User Guide.
    ///
    /// - Parameter DescribeInstanceAccessControlAttributeConfigurationInput : [no documentation found]
    ///
    /// - Returns: `DescribeInstanceAccessControlAttributeConfigurationOutputResponse` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : You do not have sufficient access to perform this action.
    /// - `InternalServerException` : The request processing has failed because of an unknown error, exception, or failure with an internal server.
    /// - `ResourceNotFoundException` : Indicates that a requested resource is not found.
    /// - `ThrottlingException` : Indicates that the principal has crossed the throttling limits of the API operations.
    /// - `ValidationException` : The request failed because it contains a syntax error.
    func describeInstanceAccessControlAttributeConfiguration(input: DescribeInstanceAccessControlAttributeConfigurationInput) async throws -> DescribeInstanceAccessControlAttributeConfigurationOutputResponse
    /// Gets the details of the permission set.
    ///
    /// - Parameter DescribePermissionSetInput : [no documentation found]
    ///
    /// - Returns: `DescribePermissionSetOutputResponse` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : You do not have sufficient access to perform this action.
    /// - `InternalServerException` : The request processing has failed because of an unknown error, exception, or failure with an internal server.
    /// - `ResourceNotFoundException` : Indicates that a requested resource is not found.
    /// - `ThrottlingException` : Indicates that the principal has crossed the throttling limits of the API operations.
    /// - `ValidationException` : The request failed because it contains a syntax error.
    func describePermissionSet(input: DescribePermissionSetInput) async throws -> DescribePermissionSetOutputResponse
    /// Describes the status for the given permission set provisioning request.
    ///
    /// - Parameter DescribePermissionSetProvisioningStatusInput : [no documentation found]
    ///
    /// - Returns: `DescribePermissionSetProvisioningStatusOutputResponse` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : You do not have sufficient access to perform this action.
    /// - `InternalServerException` : The request processing has failed because of an unknown error, exception, or failure with an internal server.
    /// - `ResourceNotFoundException` : Indicates that a requested resource is not found.
    /// - `ThrottlingException` : Indicates that the principal has crossed the throttling limits of the API operations.
    /// - `ValidationException` : The request failed because it contains a syntax error.
    func describePermissionSetProvisioningStatus(input: DescribePermissionSetProvisioningStatusInput) async throws -> DescribePermissionSetProvisioningStatusOutputResponse
    /// Detaches the specified customer managed policy from the specified [PermissionSet].
    ///
    /// - Parameter DetachCustomerManagedPolicyReferenceFromPermissionSetInput : [no documentation found]
    ///
    /// - Returns: `DetachCustomerManagedPolicyReferenceFromPermissionSetOutputResponse` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : You do not have sufficient access to perform this action.
    /// - `ConflictException` : Occurs when a conflict with a previous successful write is detected. This generally occurs when the previous write did not have time to propagate to the host serving the current request. A retry (with appropriate backoff logic) is the recommended response to this exception.
    /// - `InternalServerException` : The request processing has failed because of an unknown error, exception, or failure with an internal server.
    /// - `ResourceNotFoundException` : Indicates that a requested resource is not found.
    /// - `ThrottlingException` : Indicates that the principal has crossed the throttling limits of the API operations.
    /// - `ValidationException` : The request failed because it contains a syntax error.
    func detachCustomerManagedPolicyReferenceFromPermissionSet(input: DetachCustomerManagedPolicyReferenceFromPermissionSetInput) async throws -> DetachCustomerManagedPolicyReferenceFromPermissionSetOutputResponse
    /// Detaches the attached Amazon Web Services managed policy ARN from the specified permission set.
    ///
    /// - Parameter DetachManagedPolicyFromPermissionSetInput : [no documentation found]
    ///
    /// - Returns: `DetachManagedPolicyFromPermissionSetOutputResponse` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : You do not have sufficient access to perform this action.
    /// - `ConflictException` : Occurs when a conflict with a previous successful write is detected. This generally occurs when the previous write did not have time to propagate to the host serving the current request. A retry (with appropriate backoff logic) is the recommended response to this exception.
    /// - `InternalServerException` : The request processing has failed because of an unknown error, exception, or failure with an internal server.
    /// - `ResourceNotFoundException` : Indicates that a requested resource is not found.
    /// - `ThrottlingException` : Indicates that the principal has crossed the throttling limits of the API operations.
    /// - `ValidationException` : The request failed because it contains a syntax error.
    func detachManagedPolicyFromPermissionSet(input: DetachManagedPolicyFromPermissionSetInput) async throws -> DetachManagedPolicyFromPermissionSetOutputResponse
    /// Obtains the inline policy assigned to the permission set.
    ///
    /// - Parameter GetInlinePolicyForPermissionSetInput : [no documentation found]
    ///
    /// - Returns: `GetInlinePolicyForPermissionSetOutputResponse` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : You do not have sufficient access to perform this action.
    /// - `InternalServerException` : The request processing has failed because of an unknown error, exception, or failure with an internal server.
    /// - `ResourceNotFoundException` : Indicates that a requested resource is not found.
    /// - `ThrottlingException` : Indicates that the principal has crossed the throttling limits of the API operations.
    /// - `ValidationException` : The request failed because it contains a syntax error.
    func getInlinePolicyForPermissionSet(input: GetInlinePolicyForPermissionSetInput) async throws -> GetInlinePolicyForPermissionSetOutputResponse
    /// Obtains the permissions boundary for a specified [PermissionSet].
    ///
    /// - Parameter GetPermissionsBoundaryForPermissionSetInput : [no documentation found]
    ///
    /// - Returns: `GetPermissionsBoundaryForPermissionSetOutputResponse` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : You do not have sufficient access to perform this action.
    /// - `InternalServerException` : The request processing has failed because of an unknown error, exception, or failure with an internal server.
    /// - `ResourceNotFoundException` : Indicates that a requested resource is not found.
    /// - `ThrottlingException` : Indicates that the principal has crossed the throttling limits of the API operations.
    /// - `ValidationException` : The request failed because it contains a syntax error.
    func getPermissionsBoundaryForPermissionSet(input: GetPermissionsBoundaryForPermissionSetInput) async throws -> GetPermissionsBoundaryForPermissionSetOutputResponse
    /// Lists the status of the Amazon Web Services account assignment creation requests for a specified IAM Identity Center instance.
    ///
    /// - Parameter ListAccountAssignmentCreationStatusInput : [no documentation found]
    ///
    /// - Returns: `ListAccountAssignmentCreationStatusOutputResponse` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : You do not have sufficient access to perform this action.
    /// - `InternalServerException` : The request processing has failed because of an unknown error, exception, or failure with an internal server.
    /// - `ResourceNotFoundException` : Indicates that a requested resource is not found.
    /// - `ThrottlingException` : Indicates that the principal has crossed the throttling limits of the API operations.
    /// - `ValidationException` : The request failed because it contains a syntax error.
    func listAccountAssignmentCreationStatus(input: ListAccountAssignmentCreationStatusInput) async throws -> ListAccountAssignmentCreationStatusOutputResponse
    /// Lists the status of the Amazon Web Services account assignment deletion requests for a specified IAM Identity Center instance.
    ///
    /// - Parameter ListAccountAssignmentDeletionStatusInput : [no documentation found]
    ///
    /// - Returns: `ListAccountAssignmentDeletionStatusOutputResponse` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : You do not have sufficient access to perform this action.
    /// - `InternalServerException` : The request processing has failed because of an unknown error, exception, or failure with an internal server.
    /// - `ResourceNotFoundException` : Indicates that a requested resource is not found.
    /// - `ThrottlingException` : Indicates that the principal has crossed the throttling limits of the API operations.
    /// - `ValidationException` : The request failed because it contains a syntax error.
    func listAccountAssignmentDeletionStatus(input: ListAccountAssignmentDeletionStatusInput) async throws -> ListAccountAssignmentDeletionStatusOutputResponse
    /// Lists the assignee of the specified Amazon Web Services account with the specified permission set.
    ///
    /// - Parameter ListAccountAssignmentsInput : [no documentation found]
    ///
    /// - Returns: `ListAccountAssignmentsOutputResponse` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : You do not have sufficient access to perform this action.
    /// - `InternalServerException` : The request processing has failed because of an unknown error, exception, or failure with an internal server.
    /// - `ResourceNotFoundException` : Indicates that a requested resource is not found.
    /// - `ThrottlingException` : Indicates that the principal has crossed the throttling limits of the API operations.
    /// - `ValidationException` : The request failed because it contains a syntax error.
    func listAccountAssignments(input: ListAccountAssignmentsInput) async throws -> ListAccountAssignmentsOutputResponse
    /// Lists all the Amazon Web Services accounts where the specified permission set is provisioned.
    ///
    /// - Parameter ListAccountsForProvisionedPermissionSetInput : [no documentation found]
    ///
    /// - Returns: `ListAccountsForProvisionedPermissionSetOutputResponse` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : You do not have sufficient access to perform this action.
    /// - `InternalServerException` : The request processing has failed because of an unknown error, exception, or failure with an internal server.
    /// - `ResourceNotFoundException` : Indicates that a requested resource is not found.
    /// - `ThrottlingException` : Indicates that the principal has crossed the throttling limits of the API operations.
    /// - `ValidationException` : The request failed because it contains a syntax error.
    func listAccountsForProvisionedPermissionSet(input: ListAccountsForProvisionedPermissionSetInput) async throws -> ListAccountsForProvisionedPermissionSetOutputResponse
    /// Lists all customer managed policies attached to a specified [PermissionSet].
    ///
    /// - Parameter ListCustomerManagedPolicyReferencesInPermissionSetInput : [no documentation found]
    ///
    /// - Returns: `ListCustomerManagedPolicyReferencesInPermissionSetOutputResponse` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : You do not have sufficient access to perform this action.
    /// - `InternalServerException` : The request processing has failed because of an unknown error, exception, or failure with an internal server.
    /// - `ResourceNotFoundException` : Indicates that a requested resource is not found.
    /// - `ThrottlingException` : Indicates that the principal has crossed the throttling limits of the API operations.
    /// - `ValidationException` : The request failed because it contains a syntax error.
    func listCustomerManagedPolicyReferencesInPermissionSet(input: ListCustomerManagedPolicyReferencesInPermissionSetInput) async throws -> ListCustomerManagedPolicyReferencesInPermissionSetOutputResponse
    /// Lists the IAM Identity Center instances that the caller has access to.
    ///
    /// - Parameter ListInstancesInput : [no documentation found]
    ///
    /// - Returns: `ListInstancesOutputResponse` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : You do not have sufficient access to perform this action.
    /// - `InternalServerException` : The request processing has failed because of an unknown error, exception, or failure with an internal server.
    /// - `ThrottlingException` : Indicates that the principal has crossed the throttling limits of the API operations.
    /// - `ValidationException` : The request failed because it contains a syntax error.
    func listInstances(input: ListInstancesInput) async throws -> ListInstancesOutputResponse
    /// Lists the Amazon Web Services managed policy that is attached to a specified permission set.
    ///
    /// - Parameter ListManagedPoliciesInPermissionSetInput : [no documentation found]
    ///
    /// - Returns: `ListManagedPoliciesInPermissionSetOutputResponse` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : You do not have sufficient access to perform this action.
    /// - `InternalServerException` : The request processing has failed because of an unknown error, exception, or failure with an internal server.
    /// - `ResourceNotFoundException` : Indicates that a requested resource is not found.
    /// - `ThrottlingException` : Indicates that the principal has crossed the throttling limits of the API operations.
    /// - `ValidationException` : The request failed because it contains a syntax error.
    func listManagedPoliciesInPermissionSet(input: ListManagedPoliciesInPermissionSetInput) async throws -> ListManagedPoliciesInPermissionSetOutputResponse
    /// Lists the status of the permission set provisioning requests for a specified IAM Identity Center instance.
    ///
    /// - Parameter ListPermissionSetProvisioningStatusInput : [no documentation found]
    ///
    /// - Returns: `ListPermissionSetProvisioningStatusOutputResponse` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : You do not have sufficient access to perform this action.
    /// - `InternalServerException` : The request processing has failed because of an unknown error, exception, or failure with an internal server.
    /// - `ResourceNotFoundException` : Indicates that a requested resource is not found.
    /// - `ThrottlingException` : Indicates that the principal has crossed the throttling limits of the API operations.
    /// - `ValidationException` : The request failed because it contains a syntax error.
    func listPermissionSetProvisioningStatus(input: ListPermissionSetProvisioningStatusInput) async throws -> ListPermissionSetProvisioningStatusOutputResponse
    /// Lists the [PermissionSet]s in an IAM Identity Center instance.
    ///
    /// - Parameter ListPermissionSetsInput : [no documentation found]
    ///
    /// - Returns: `ListPermissionSetsOutputResponse` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : You do not have sufficient access to perform this action.
    /// - `InternalServerException` : The request processing has failed because of an unknown error, exception, or failure with an internal server.
    /// - `ResourceNotFoundException` : Indicates that a requested resource is not found.
    /// - `ThrottlingException` : Indicates that the principal has crossed the throttling limits of the API operations.
    /// - `ValidationException` : The request failed because it contains a syntax error.
    func listPermissionSets(input: ListPermissionSetsInput) async throws -> ListPermissionSetsOutputResponse
    /// Lists all the permission sets that are provisioned to a specified Amazon Web Services account.
    ///
    /// - Parameter ListPermissionSetsProvisionedToAccountInput : [no documentation found]
    ///
    /// - Returns: `ListPermissionSetsProvisionedToAccountOutputResponse` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : You do not have sufficient access to perform this action.
    /// - `InternalServerException` : The request processing has failed because of an unknown error, exception, or failure with an internal server.
    /// - `ResourceNotFoundException` : Indicates that a requested resource is not found.
    /// - `ThrottlingException` : Indicates that the principal has crossed the throttling limits of the API operations.
    /// - `ValidationException` : The request failed because it contains a syntax error.
    func listPermissionSetsProvisionedToAccount(input: ListPermissionSetsProvisionedToAccountInput) async throws -> ListPermissionSetsProvisionedToAccountOutputResponse
    /// Lists the tags that are attached to a specified resource.
    ///
    /// - Parameter ListTagsForResourceInput : [no documentation found]
    ///
    /// - Returns: `ListTagsForResourceOutputResponse` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : You do not have sufficient access to perform this action.
    /// - `InternalServerException` : The request processing has failed because of an unknown error, exception, or failure with an internal server.
    /// - `ResourceNotFoundException` : Indicates that a requested resource is not found.
    /// - `ThrottlingException` : Indicates that the principal has crossed the throttling limits of the API operations.
    /// - `ValidationException` : The request failed because it contains a syntax error.
    func listTagsForResource(input: ListTagsForResourceInput) async throws -> ListTagsForResourceOutputResponse
    /// The process by which a specified permission set is provisioned to the specified target.
    ///
    /// - Parameter ProvisionPermissionSetInput : [no documentation found]
    ///
    /// - Returns: `ProvisionPermissionSetOutputResponse` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : You do not have sufficient access to perform this action.
    /// - `ConflictException` : Occurs when a conflict with a previous successful write is detected. This generally occurs when the previous write did not have time to propagate to the host serving the current request. A retry (with appropriate backoff logic) is the recommended response to this exception.
    /// - `InternalServerException` : The request processing has failed because of an unknown error, exception, or failure with an internal server.
    /// - `ResourceNotFoundException` : Indicates that a requested resource is not found.
    /// - `ThrottlingException` : Indicates that the principal has crossed the throttling limits of the API operations.
    /// - `ValidationException` : The request failed because it contains a syntax error.
    func provisionPermissionSet(input: ProvisionPermissionSetInput) async throws -> ProvisionPermissionSetOutputResponse
    /// Attaches an inline policy to a permission set. If the permission set is already referenced by one or more account assignments, you will need to call [ProvisionPermissionSet] after this action to apply the corresponding IAM policy updates to all assigned accounts.
    ///
    /// - Parameter PutInlinePolicyToPermissionSetInput : [no documentation found]
    ///
    /// - Returns: `PutInlinePolicyToPermissionSetOutputResponse` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : You do not have sufficient access to perform this action.
    /// - `ConflictException` : Occurs when a conflict with a previous successful write is detected. This generally occurs when the previous write did not have time to propagate to the host serving the current request. A retry (with appropriate backoff logic) is the recommended response to this exception.
    /// - `InternalServerException` : The request processing has failed because of an unknown error, exception, or failure with an internal server.
    /// - `ResourceNotFoundException` : Indicates that a requested resource is not found.
    /// - `ServiceQuotaExceededException` : Indicates that the principal has crossed the permitted number of resources that can be created.
    /// - `ThrottlingException` : Indicates that the principal has crossed the throttling limits of the API operations.
    /// - `ValidationException` : The request failed because it contains a syntax error.
    func putInlinePolicyToPermissionSet(input: PutInlinePolicyToPermissionSetInput) async throws -> PutInlinePolicyToPermissionSetOutputResponse
    /// Attaches an Amazon Web Services managed or customer managed policy to the specified [PermissionSet] as a permissions boundary.
    ///
    /// - Parameter PutPermissionsBoundaryToPermissionSetInput : [no documentation found]
    ///
    /// - Returns: `PutPermissionsBoundaryToPermissionSetOutputResponse` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : You do not have sufficient access to perform this action.
    /// - `ConflictException` : Occurs when a conflict with a previous successful write is detected. This generally occurs when the previous write did not have time to propagate to the host serving the current request. A retry (with appropriate backoff logic) is the recommended response to this exception.
    /// - `InternalServerException` : The request processing has failed because of an unknown error, exception, or failure with an internal server.
    /// - `ResourceNotFoundException` : Indicates that a requested resource is not found.
    /// - `ThrottlingException` : Indicates that the principal has crossed the throttling limits of the API operations.
    /// - `ValidationException` : The request failed because it contains a syntax error.
    func putPermissionsBoundaryToPermissionSet(input: PutPermissionsBoundaryToPermissionSetInput) async throws -> PutPermissionsBoundaryToPermissionSetOutputResponse
    /// Associates a set of tags with a specified resource.
    ///
    /// - Parameter TagResourceInput : [no documentation found]
    ///
    /// - Returns: `TagResourceOutputResponse` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : You do not have sufficient access to perform this action.
    /// - `ConflictException` : Occurs when a conflict with a previous successful write is detected. This generally occurs when the previous write did not have time to propagate to the host serving the current request. A retry (with appropriate backoff logic) is the recommended response to this exception.
    /// - `InternalServerException` : The request processing has failed because of an unknown error, exception, or failure with an internal server.
    /// - `ResourceNotFoundException` : Indicates that a requested resource is not found.
    /// - `ServiceQuotaExceededException` : Indicates that the principal has crossed the permitted number of resources that can be created.
    /// - `ThrottlingException` : Indicates that the principal has crossed the throttling limits of the API operations.
    /// - `ValidationException` : The request failed because it contains a syntax error.
    func tagResource(input: TagResourceInput) async throws -> TagResourceOutputResponse
    /// Disassociates a set of tags from a specified resource.
    ///
    /// - Parameter UntagResourceInput : [no documentation found]
    ///
    /// - Returns: `UntagResourceOutputResponse` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : You do not have sufficient access to perform this action.
    /// - `ConflictException` : Occurs when a conflict with a previous successful write is detected. This generally occurs when the previous write did not have time to propagate to the host serving the current request. A retry (with appropriate backoff logic) is the recommended response to this exception.
    /// - `InternalServerException` : The request processing has failed because of an unknown error, exception, or failure with an internal server.
    /// - `ResourceNotFoundException` : Indicates that a requested resource is not found.
    /// - `ThrottlingException` : Indicates that the principal has crossed the throttling limits of the API operations.
    /// - `ValidationException` : The request failed because it contains a syntax error.
    func untagResource(input: UntagResourceInput) async throws -> UntagResourceOutputResponse
    /// Updates the IAM Identity Center identity store attributes that you can use with the IAM Identity Center instance for attributes-based access control (ABAC). When using an external identity provider as an identity source, you can pass attributes through the SAML assertion as an alternative to configuring attributes from the IAM Identity Center identity store. If a SAML assertion passes any of these attributes, IAM Identity Center replaces the attribute value with the value from the IAM Identity Center identity store. For more information about ABAC, see [Attribute-Based Access Control] in the IAM Identity Center User Guide.
    ///
    /// - Parameter UpdateInstanceAccessControlAttributeConfigurationInput : [no documentation found]
    ///
    /// - Returns: `UpdateInstanceAccessControlAttributeConfigurationOutputResponse` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : You do not have sufficient access to perform this action.
    /// - `ConflictException` : Occurs when a conflict with a previous successful write is detected. This generally occurs when the previous write did not have time to propagate to the host serving the current request. A retry (with appropriate backoff logic) is the recommended response to this exception.
    /// - `InternalServerException` : The request processing has failed because of an unknown error, exception, or failure with an internal server.
    /// - `ResourceNotFoundException` : Indicates that a requested resource is not found.
    /// - `ThrottlingException` : Indicates that the principal has crossed the throttling limits of the API operations.
    /// - `ValidationException` : The request failed because it contains a syntax error.
    func updateInstanceAccessControlAttributeConfiguration(input: UpdateInstanceAccessControlAttributeConfigurationInput) async throws -> UpdateInstanceAccessControlAttributeConfigurationOutputResponse
    /// Updates an existing permission set.
    ///
    /// - Parameter UpdatePermissionSetInput : [no documentation found]
    ///
    /// - Returns: `UpdatePermissionSetOutputResponse` : [no documentation found]
    ///
    /// - Throws: One of the exceptions listed below __Possible Exceptions__.
    ///
    /// __Possible Exceptions:__
    /// - `AccessDeniedException` : You do not have sufficient access to perform this action.
    /// - `ConflictException` : Occurs when a conflict with a previous successful write is detected. This generally occurs when the previous write did not have time to propagate to the host serving the current request. A retry (with appropriate backoff logic) is the recommended response to this exception.
    /// - `InternalServerException` : The request processing has failed because of an unknown error, exception, or failure with an internal server.
    /// - `ResourceNotFoundException` : Indicates that a requested resource is not found.
    /// - `ThrottlingException` : Indicates that the principal has crossed the throttling limits of the API operations.
    /// - `ValidationException` : The request failed because it contains a syntax error.
    func updatePermissionSet(input: UpdatePermissionSetInput) async throws -> UpdatePermissionSetOutputResponse
}

public enum SSOAdminClientTypes {}
